persistence/src/main/java/de/cyface/persistence/serialization/MeasurementSerializer.java

```diff
@@ -356,6 +355,31 @@ public void loadSerialized(@NonNull final BufferedOutputStream bufferedOutputStr
         final File directionFile = fileAccessLayer.getFilePath(persistence.getContext(), measurementIdentifier,
                 Point3dFile.DIRECTIONS_FOLDER_NAME, Point3dFile.DIRECTION_FILE_EXTENSION);
 
+        // Calculate how many points the files contain (for the binary header)
+        int accelerationsCount = 0;
+        int rotationsCount = 0;
+        int directionsCount = 0;
+        // noinspection ConstantConditions // can happen in tests
+        if (accelerationFile != null && accelerationFile.exists()) {
+            accelerationsCount = (int)(accelerationFile.length() / BYTES_IN_ONE_POINT_3D_ENTRY);
+            Validate.isTrue(accelerationsCount * BYTES_IN_ONE_POINT_3D_ENTRY == accelerationFile.length());
```

Comment:
To make sure there is no "half" point entry at the end (i.e. bytes % bytes_per_point == 0)

This way we see if this happens at all so that we need to handle this.